documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -194,28 +638,92 @@ The auto config properties for distributed caches are:
 [options="header",cols="1,3,10"]
 |==============================================================================
 | Property name | value | description
-| hibernate.search.default.directory_provider | infinispan | Indexes stored in Infinispan. More details at link:http://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#infinispan-directories[Hibernate Search documentation]
-| hibernate.search.default.exclusive_index_use | true | indexing operation in exclusive mode, allowing Hibernate Search to optimize writes
-| hibernate.search.default.indexmanager | org.infinispan.query.indexmanager.InfinispanIndexManager | Delegates index writing to a single node in the Infinispan cluster
-| hibernate.search.default.reader.strategy | shared | Reuse index reader across several queries, avoiding reopening it
+| default.directory_provider | infinispan | Indexes stored in Infinispan. More details at link:http://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#infinispan-directories[Hibernate Search documentation]
+| default.exclusive_index_use | true | indexing operation in exclusive mode, allowing Hibernate Search to optimize writes
+| default.indexmanager | org.infinispan.query.indexmanager.InfinispanIndexManager | Delegates index writing to a single node in the Infinispan cluster
+| default.reader.strategy | shared | Reuse index reader across several queries, avoiding reopening it
 |==============================================================================
 
+===== Re-indexing [[query.massindexer]]
+
+Occasionally you might need to rebuild the Lucene index by reconstructing it from the data stored in the Cache. You need to rebuild the index if you change the definition of what is indexed on your types, or if you change for example some _Analyzer_ parameter, as Analyzers affect how the index is defined. Also, you might need to rebuild the index if you had it destroyed by some system administration mistake. To rebuild the index just get a reference to the MassIndexer and start it; beware if might take some time as it needs to reprocess all data in the grid!
```

Comment:
Please add some notice like: "We recommend you test a disaster recovery plan including running the MassIndexer. Understanding its working and tuning options in advance can make it much faster, and nobody wants to find out that their recovery process is going to take a very long time during an emergency: make sure it's tuned correctly so you can perform disaster recovery in short time."